wls-auth-service/src/test/java/de/muenchen/oss/wahllokalsystem/authservice/rest/WahllokalBenutzerRestControllerIntegrationTest.java

```diff
@@ -0,0 +1,189 @@
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import org.springframework.transaction.support.TransactionTemplate;
+
+@SpringBootTest(
+        classes = MicroServiceApplication.class,
+        properties = { "serviceauth.crypto.key=secret", "serviceauth.user.authority.wahlvorstand=" + PROP_USER_AUTHORITY_WAHLVORSTAND }
+)
+@AutoConfigureMockMvc
+@ActiveProfiles({ TestConstants.SPRING_TEST_PROFILE, TestConstants.SPRING_NO_SECURITY_PROFILE })
+class WahllokalBenutzerRestControllerIntegrationTest {
+
+    public static final String PROP_USER_AUTHORITY_WAHLVORSTAND = "WLS_USER_AUTHORITY_WAHLVORSTAND";
+
+    @Autowired
+    EntityManager entityManager;
+
+    @Autowired
+    UserRepository userRepository;
+
+    @Autowired
+    AuthorityRepository authorityRepository;
+
+    @Autowired
+    PermissionRepository permissionRepository;
+
+    @Autowired
+    MockMvc mockMvc;
+
+    @Autowired
+    ObjectMapper objectMapper;
+
+    @Autowired
+    TransactionTemplate transactionTemplate;
+
+    @Autowired
+    CryptoService cryptoService;
+
+    @AfterEach
+    void tearDown() {
+        transactionTemplate.executeWithoutResult(status -> entityManager.createQuery("DELETE FROM User").executeUpdate());
+        permissionRepository.deleteAll();
+        authorityRepository.deleteAll();
+    }
+
+    @Nested
+    class CreateAndExportWahllokalBenutzer {
+
+        @Test
+        void should_persistUsers_when_usersAreGiven() throws Exception {
+            val wahltagID = "wahltagID";
+            val wahltag = LocalDate.now();
+            val user1 = new WahllokalUserInfoDTO("1", wahltag, "wbzID1", WahlbezirksartDTO.UWB, "1_1");
+            val user2 = new WahllokalUserInfoDTO("2", wahltag, "wbzID2", WahlbezirksartDTO.BWB, "2_1");
+            val requestBody = List.of(user1, user2);
+
+            authorityRepository.save(new Authority(PROP_USER_AUTHORITY_WAHLVORSTAND, Collections.emptySet(), Collections.emptySet()));
+
+            val request = MockMvcRequestBuilders.post("/generateAndExportWahllokalbenutzer/" + wahltagID).contentType(MediaType.APPLICATION_JSON)
+                    .content(objectMapper.writeValueAsBytes(requestBody));
+
+            val result = mockMvc.perform(request).andExpect(status().isCreated()).andReturn();
+
+            final List<User> persistedUsers = (List) entityManager.createQuery("SELECT u FROM User u").getResultList();
+            val username1 = cryptoService.decrypt(persistedUsers.get(0).getUsername());
+            val username2 = cryptoService.decrypt(persistedUsers.get(1).getUsername());
+            val expectedResult = username1 + "\r\n" + username2;
+
+            Assertions.assertThat(persistedUsers).hasSize(2);
+            Assertions.assertThat(result.getResponse().getContentAsString()).isEqualTo(expectedResult);
+        }
+
+        @Test
+        void should_replaceOldUsers_when_usersAreGiven() throws Exception {
```

Comment:
```suggestion
        void should_deleteOldUsersAndPersistNewUsers_when_UsersAreGiven() throws Exception {
```
Und dann vielleicht noch einen Test, der zeigt, dass dies nur für die Benutzer mit einer bestimmten `wahltagID` gilt um die "Multiwahlfunktionalität" zu testen?

Reply:
habe den bestehenden Test angepasst. Ich denke es passt gut rein: 29344583895df09f4f124487f488e3ccc816a8f5